src/main/groovy/net/researchgate/release/ReleasePlugin.groovy

```diff
@@ -157,18 +164,40 @@ class ReleasePlugin extends PluginHelper implements Plugin<Project> {
             }
         }
 
-        project.gradle.taskGraph.afterTask { Task task, TaskState state ->
-            if (state.failure && task.name == "release") {
-                try {
-                    createScmAdapter()
-                } catch (Exception ignored) {}
-                if (scmAdapter && extension.revertOnFail && project.file(extension.versionPropertyFile)?.exists()) {
-                    log.error('Release process failed, reverting back any changes made by Release Plugin.')
-                    scmAdapter.revert()
-                } else {
-                    log.error('Release process failed, please remember to revert any uncommitted changes made by the Release Plugin.')
+        if (GradleVersion.current() < GradleVersion.version('6.1')) {
+            project.gradle.taskGraph.afterTask { Task task, TaskState state ->
+                if (state.failure && task.name == 'release') {
+                    revert()
                 }
             }
+        } else {
+            objects.newInstance(BuildEventsListenerRegistryProvider)
+                  .buildEventsListenerRegistry
+                  .onTaskCompletion(providers.provider {
+                      { finishEvent ->
+                          if ((finishEvent.result instanceof TaskFailureResult) && finishEvent.descriptor.taskPath.endsWith(':release')) {
+                              revert()
+                          }
+                      } as OperationCompletionListener
+                  })
+        }
+    }
+
+    @Inject
+    abstract protected ObjectFactory getObjects();
+
+    @Inject
+    abstract protected ProviderFactory getProviders();
+
+    protected revert() {
+        try {
+            createScmAdapter()
+        } catch (Exception ignored) {}
+        if (scmAdapter && extension.revertOnFail && project.file(extension.versionPropertyFile)?.exists()) {
```

Comment:
```suggestion
        if (scmAdapter && extension.revertOnFail.get() && project.file(extension.versionPropertyFile)?.exists()) {
```
Not sure what groovy does if we don't use the get() .

Reply:
The same it does currently, I just moved this existing code to a method.
But you are fully right of course, for Groovy it will just always be `true` as it is non-null.
This is already like that since switching to properties and not adapting this usage (and maybe others?)
One of the fun things you have because of using Groovy for the plugin. :-D

I now fixed it together with this PR.